game/view/readyability.lua

```diff
@@ -0,0 +1,232 @@
+
+local RES = require 'resources'
+local Color = require 'common.color'
+local DEFS = require 'domain.definitions'
+local COLORS = require 'domain.definitions.colors'
+local FONT = require 'view.helpers.font'
+
+local min = math.min
+local abs = math.abs
+local floor = math.floor
+
+local ReadyAbilityView = Class{
+  __includes = { ELEMENT }
+}
+
+local _FADE_TIMER = "FADE_TIMER"
+local _LIST_TIMER = "LIST_TIMER"
+local _OFFSET_TIMER = "READY_ABILITY_OFFSET_TIMER"
+local _MOVE_SPEED = 0.25
+
+local _FONT_NAME = "Text"
+local _FONT_SIZE = 20
+local _MARGIN = 8
+local _PADDING = 8
+local _FMT = "%s (%d)"
+local _TRIGGERS = {
+  [DEFS.TRIGGERS.ON_USE] = " uses",
+  [DEFS.TRIGGERS.ON_TURN] = " turns",
+  [DEFS.TRIGGERS.ON_TICK] = " ticks",
+}
+
+local _WIDTH, _HEIGHT
+local _font
+local _alpha
+
+local function _initGraphicValues()
+  local g = love.graphics
+  _WIDTH, _HEIGHT = g.getDimensions()
+  _font = _font or FONT.get(_FONT_NAME, _FONT_SIZE)
+  _font:setFilter('linear', 'linear', 1)
+end
+
+local function _next(i, len, n)
+  if n == 0 then return i end
+  return _next(i % len + 1, len, n - 1)
+end
+
+local function _prev(i, len, n)
+  if n == 0 then return i end
+  return _prev((i - 2) % len + 1, len, n - 1)
+end
+
+local function _dist(i, j, len)
+  local d = i - j
+  if d > len/2 then
+    d = d - len
+  elseif d < -len/2 then
+    d = d + len
+  end
+  return d
+end
+
+local function _drawContainer(g, width)
+  local h = _font:getHeight() + _PADDING*2 + 2*_MARGIN + 4
+  g.rectangle("fill",
+    -_MARGIN*4,
+    -h - _MARGIN,
+    width + 4*_PADDING + _MARGIN*8,
+    h*3
+  )
+end
+
+function ReadyAbilityView:init(widgets, selection)
+  ELEMENT.init(self)
+
+  self.widgets = widgets
+  self.widget_count = #widgets
+  self.selection = selection
+  self.offset = 0
+  self.alpha = 0
+  self.list_alpha = 0
+
+  _initGraphicValues()
+end
+
+function ReadyAbilityView:setSelection(n)
+  self.selection = n
+end
+
+function ReadyAbilityView:getSelection()
+  return self.selection
+end
+
+function ReadyAbilityView:selectNext()
+  local previous = self.selection
+  self.selection = _next(self.selection, self.widget_count, 1)
+  self.offset = _dist(self.selection, previous, self.widget_count)
```

Comment:
If you do
```lua
-- in selectNext()
self.offset = math.abs(_dist(...))
-- in selectPrev()
self.offset = -math.abs(_dist(...))
```

Then it properly animates for two selectable abilities. It assumes you never move more than 1 at a time, which seems reasonable given how this is coded.

Reply:
Actually, you can just do
```lua
self.offset = 1 -- in selectNext
self.offset = -1 -- in selectPrev
```
Because the line before you explicitly move `self.selection` by either +1 or -1.